chapters/scoping.tex

```diff
@@ -91,8 +91,10 @@ \subsection{Composite Name Lookup}\label{composite-name-lookup}
 \item
   If the first identifier denotes a component, the rest of the name (e.g., \lstinline!B! or \lstinline!B.C!) is looked up among the declared named component elements of the component.
 \item
-  If not found, and if the first identifier denotes a scalar component, or \lstinline!component[j]! where \lstinline!component! is an array of components and the indices \lstinline!j! can be evaluated at translation time and \lstinline!component[j]! is a scalar; and if the composite name is used as a function call, the lookup is also performed among the declared named class elements of the scalar component, and must find a non-operator function.
+  If not found, and if the first identifier denotes a component and the composite name is used as a function call, the lookup is also performed among the declared elements of the component, and must find a non-operator function.
+  Each leading element, including the first one, must in this case be a scalar component, or \lstinline!component[j]! where \lstinline!component! is an array of components and the indices \lstinline!j! can be evaluated at translation time and \lstinline!component[j]! is a scalar.
   All identifiers of the rest of the name (e.g., \lstinline!B! and \lstinline!B.C!) must be classes.
+  Therefore the composite name is comprised of one or more component names (optionally with indexing), followed by one or more class names.
```

Comment:
Isn't this an odd way to use _Therefore_?  What is the sentence really trying to say?

Reply:
It explains the consequences of the previous text to further clarify it. I'm not a native speaker, but I don't find it odd at all.

Reply:
In that case, I would find it more natural to express it this way:
```suggestion
  That is, the composite name is comprised of one or more component names (optionally with indexing), followed by one or more class names.
```

Reply:
Ok, makes sense.